ledger.go

```diff
@@ -1714,3 +1669,64 @@ func (l *Ledger) storeTransactions(stored *Block, evicted *Block) error {
 
 	return nil
 }
+
+// filterInvalidVotes takes a slice of (*finalizationVote)'s and filters away
+// ones that are invalid with respect to the current nodes state.
+func (l *Ledger) filterInvalidVotes(current *Block, votes []Vote) {
+ValidateVotes:
+	for _, vote := range votes {
+		vote := vote.(*finalizationVote)
+
+		// Ignore nil block proposals.
+		if vote.block == nil {
+			continue ValidateVotes
+		}
+
+		// Ignore block proposals at an unexpected height.
+		if vote.block.Index != current.Index+1 {
+			vote.block = nil
+			continue ValidateVotes
+		}
+
+		// Ignore block proposals containing transactions which our node has not
+		// locally archived, and mark them as missing.
+		if l.transactions.BatchMarkMissing(vote.block.Transactions...) {
+			vote.block = nil
+			continue ValidateVotes
+		}
+
+		transactions, err := l.transactions.BatchFind(vote.block.Transactions)
+		if err != nil {
+			vote.block = nil
+			continue ValidateVotes
+		}
+
+		// Validate the height recorded on transactions inside the block proposal.
+		for _, tx := range transactions {
+			if vote.block.Index >= tx.Block+uint64(conf.GetPruningLimit()) {
+				vote.block = nil
+				continue ValidateVotes
```

Comment:
thats the only place where you need to use continue to label, in all other places its one for loop and just continue can be used.
also, maybe instead of nested loop and label separate function with return can be used?

Reply:
Hm, I think it would be a bit odd to separate `filterInvalidVotes` into two separate functions where one of the functions gets called on each vote and nullifies `vote.block`. Hence why I kept it to just one single function and labeled the loop instead.